### PR TITLE
Addressed issue with min/max levels. Win rate is up but is currently pro...

### PR DESCRIPTION
**Pull request by Randy-Mills**
...ne to server timeouts when the timer is set to 10 seconds. Next steps are to avoid redundant computation. IE stop searching when we have found a win, avoid visiting nodes multiple times.

### CODE REVIEW

File: src/QuartoGameState.java [hunk 1]
```diff
@@ -103,36 +103,19 @@ public int calcNodesInGeneration(int generations) {
...
-            this.board.printBoardState();
-            System.out.println("\n");
+            value = isMaxState ? -27 : 27;
         }
         else {
-            System.out.println("\nStarting board state");
-            this.board.printBoardState();
-            System.out.println("\n");
-
-            // FIXME/OPTIMIZE
```
Comment (rkillam) [inline src/QuartoGameState.java:118]: Should still be optimized, or at least de-hackyfied. Having a second hasQuarto method is pretty hacky.

File: src/QuartoPlayerAgent.java [hunk 2]
```diff
@@ -140,13 +134,14 @@ else if(state.value == curState.value) {
      */
     @Override
     protected String moveSelectionAlgorithm(int pieceID) {
-        QuartoGameState tmpState = new QuartoGameState(this.quartoBoard, Integer.MAX_VALUE, Integer.MIN_VALUE, true);
+        /*
+         * NOTE: the new game state is set as a minNode because then we pass this state to the
+         * generator to start trying to place the new piece we flip to a maxNode. This isn't
+         * intuitive to me but appears to result in a winning AI.
```
Comment (rkillam) [inline src/QuartoPlayerAgent.java:140]: YAY!!!... I'll look into this to see if we can have both an intuitive AI AND a winning AI

**Comment by rkillam** [comment]
PR approved. 

If Brenden doesn't get around to looking at this by tomorrow I'll merge it in.

**Comment by dbrenden** [comment]
Lookin' good.

**Pull request merged**